python/ray/data/impl/util.py

```diff
@@ -30,3 +35,66 @@ def _check_pyarrow_version():
                            "to ensure compatibility with Ray Datasets.")
         else:
             _VERSION_VALIDATED = True
+
+
+def _get_spread_resources_iter(nodes: List[Dict[str, Any]],
+                               spread_resource_prefix: str,
+                               ray_remote_args: Dict[str, Any]):
+    """Returns a round-robin iterator over resources that match the given
+    prefix and that coexist on nodes with the resource requests given in the
+    provided remote args (along with the task resource request defaults).
+    """
+    # Extract the resource labels from the remote args.
```

Comment:
should we sort the nodes?

Reply:
@scv119 The returned resource labels from `_filtered_resources()` are sorted!